redfish-core/lib/log_services.hpp

```diff
@@ -1085,6 +1116,40 @@ inline void requestRoutesSystemLogServiceCollection(App& app)
                     "/xyz/openbmc_project/object_mapper",
                     "xyz.openbmc_project.ObjectMapper", "GetSubTreePaths", "/",
                     0, std::array<const char*, 1>{postCodeIface});
+
+                crow::connections::systemBus->async_method_call(
+                    [asyncResp](const boost::system::error_code ec,
+                                const std::vector<std::string>& subtreePath) {
+                        if (ec)
+                        {
+                            BMCWEB_LOG_ERROR << ec;
+                            return;
+                        }
+
+                        for (auto& pathStr : subtreePath)
+                        {
+                            if (pathStr.find("hardware_isolation") !=
+                                std::string::npos)
+                            {
+                                nlohmann::json& logServiceArrayLocal =
+                                    asyncResp->res.jsonValue["Members"];
+                                logServiceArrayLocal.push_back(
+                                    {{"@odata.id",
+                                      "/redfish/v1/Systems/system/"
+                                      "LogServices/HardwareIsolation"}});
+                                asyncResp->res
+                                    .jsonValue["Members@odata.count"] =
+                                    logServiceArrayLocal.size();
+                                return;
+                            }
+                        }
+                    },
+                    "xyz.openbmc_project.ObjectMapper",
+                    "/xyz/openbmc_project/object_mapper",
+                    "xyz.openbmc_project.ObjectMapper", "GetSubTreePaths", "/",
+                    0,
+                    std::array<const char*, 1>{
+                        "xyz.openbmc_project.HardwareIsolation.Create"});
```

Comment:
Seems like an odd interface for a log service? 
xyz.openbmc_project.HardwareIsolation.Log 
xyz.openbmc_project.HardwareIsolation.Event

Reply:
I think, we need to return `"/redfish/v1/Systems/system/LogServices/HardwareIsolation"` in the LogServiceCollection when query like `curl -k -H "X-Auth-Token: $bmc_token" -X GET \
https://${bmc}/redfish/v1/Systems/system/LogServices`  based on the backend support so using the **xyz.openbmc_project.HardwareIsolation.Create** (hosted in the HardwareIsolation root D-Bus path) to decide whether the backend support is present or not since without Create API we cannot create the guard record in the D-Bus. 

The one you mentioned needs to add in D-Bus YAML to host in the HardwareIsolation root D-Bus path but, I am seeing any value to just decide the backend support for the bmcweb use case when we have other D-Bus interfaces.

Reply:
bmcweb typically looks for an "Item.Inventory" interface for inventory resource. 
It typically looks for like a "Logging" interface for Log resources. 
This just seems odd to look for "xyz.openbmc_project.HardwareIsolation.Create"